toolkit/tools/pkg/licensecheck/licensecheck.go

```diff
@@ -0,0 +1,359 @@
+	licenseNames     LicenseNames                       // The regexes used to match license files
+	exceptions       LicenseExceptions                  // Files that should be ignored
+	results          []LicenseCheckResult               // The results of the search
+	jobSemaphore     chan struct{}                      // Limit the number of parallel jobs
+}
+
+// New creates a new license checker. If this returns successfully the caller is responsible for calling CleanUp().
+// - buildDirPath: The path to create the chroot inside
+// - workerTarPath: The path to the worker tarball
+// - rpmDirPath: The path to the directory containing the RPMs
+// - nameFilePath: The path to the .json file containing license names
+// - exceptionFilePath: Optional, the path to the .json file containing license exceptions to ignore
+// - distTag: The distribution tag to use when parsing RPMs
+func New(buildDirPath, workerTarPath, rpmDirPath, nameFilePath, exceptionFilePath, distTag string,
+) (newLicenseChecker *LicenseChecker, err error) {
+	const chrootName = "license_chroot"
+
+	newLicenseChecker = &LicenseChecker{
+		distTag:          distTag,
+		simpleToolChroot: &simpletoolchroot.SimpleToolChroot{},
+		jobSemaphore:     make(chan struct{}, runtime.NumCPU()*2),
+	}
+
+	err = newLicenseChecker.simpleToolChroot.InitializeChroot(buildDirPath, chrootName, workerTarPath, rpmDirPath)
+	if err != nil {
+		err = fmt.Errorf("failed to initialize chroot:\n%w", err)
+		return nil, err
+	}
+	defer func() {
+		if err != nil {
+			cleanupErr := newLicenseChecker.Cleanup()
+			if cleanupErr != nil {
+				// Append the cleanup error to the existing error
+				err = fmt.Errorf("%w\nfailed to cleanup after failing to create a new LicenseChecker:\n%w", err, cleanupErr)
+			}
+		}
+	}()
+
+	newLicenseChecker.licenseNames, err = LoadLicenseNames(nameFilePath)
+	if err != nil {
+		err = fmt.Errorf("failed to load license names:\n%w", err)
+		return nil, err
+	}
+
+	if exceptionFilePath != "" {
+		newLicenseChecker.exceptions, err = LoadLicenseExceptions(exceptionFilePath)
+		if err != nil {
+			err = fmt.Errorf("failed to load license exceptions:\n%w", err)
+			return nil, err
+		}
+	}
+
+	return newLicenseChecker, err
+}
+
+// Cleanup tears down the chroot. If the chroot was created it will be cleaned up. Reset the struct to its initial state.
+func (l *LicenseChecker) Cleanup() error {
+	if l.simpleToolChroot != nil {
+		err := l.simpleToolChroot.CleanUp()
+		if err != nil {
+			return fmt.Errorf("failed to cleanup chroot:\n%w", err)
+		}
+		l.simpleToolChroot = nil
+	}
+	return nil
+}
+
+// CheckLicenses will scan all .rpm files in the chroot for bad licenses. New unfiltered results will be returned but
+// also appended to the internal results list which can be accessed with GetResults().
+func (l *LicenseChecker) CheckLicenses(quiet bool) (latestResults []LicenseCheckResult, err error) {
```

Comment:
Does anything ever call `CheckLicenses` with `quiet` set to `true`?

Reply:
Follow on PR uses it as part of the package builds. I don't want it polluting the log output in that use-case when we already have an in-depth results output.